### DISCUSSION TITLE
for(;;) formatted to for (; ;). Is this intentional?

### DISCUSSION THREAD

**Discussion by oriongonza** [rome/tools#3510] [Q&A]
I've honestly never seen for (; ;), and it looks kind of ugly to me. Should this be an exception?

**Comment by MichaReiser** [comment]
I'm unsure if I understand the question correctly. It would help me if you could create a [playground](https://play.rome.tools) with some sample code and share the link. 

I tried `for (;;) {}` and Rome doesn't insert a space between the two semicolons [play](https://play.rome.tools/?code=for+%28%3B%3B%29+%7B%0A++%0A++%0A%7D&lineWidth=80&indentStyle=tab&quoteStyle=double&quoteProperties=as-needed&trailingComma=all&indentWidth=2&sourceType=module&enabledNurseryRules=true&typescript=false&jsx=false#ZgBvAHIAIAAoADsAOwApACAAewAKACAAIAAKACAAIAAKAH0A)

**Reply by oriongonza** [reply]
![image](https://user-images.githubusercontent.com/47633543/198259654-f673221c-d0d7-40c1-9176-8240549648ab.png)

**Reply by MichaReiser** [reply]
Thank you. If I understand it correctly your question is why rome **is not** inserting whitespace between the `;` if the initializer, update, and test expressions are missing. 

The writing of `for (;;)` in that case is very common, and you can find an example on the [MDN website](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for#optional_for_expressions) that uses that exact formatting (unfortunately, I'm not able to directly link to the example, only the section).

**Reply by oriongonza** [reply]
Quite the opposite, rome-cli does it well (the example I gave), but the vscode rome formatter does the opposite. I might not have it properly configured.

**Reply by MichaReiser** [reply]
Hm, that's weird. My suspicion is that Rome isn't configured as the default formatter. 

Can you open the command Pallet (command + shift + P or View -> Command Pallet), then search for "Format Document With" and select Rome. If that results in the expected formatting, then change the default formatter by opening the Command Pallet again, search for "Format Document With", select "Configure Default Formatter" and then pick Rome.

Let me know if that works.

**Reply by oriongonza** [reply]
Oh that's weird, it wasn't set, now it works very well. But I set the default formatter before here, in the settings:
![image](https://user-images.githubusercontent.com/47633543/198527571-f0439a14-99d7-4a32-b5f7-7741aa054777.png)

Interesting. Thanks!
